File: Find the Town Judge.java
// https://leetcode.com/problems/find-the-town-judge/

class Solution {
    public int findJudge(int n, int[][] trust) {
        
        // Create 2 arrays - one for trust and other to see if ith trust somebody
        int[] trustCount = new int[n + 1];
        boolean[] trustAnother = new boolean[n + 1];
        
        // Iterate trust array
        for (int[] pair: trust) {
            
            // Break down pair
            int truster = pair[0];
            int trustee = pair[1];
            
            // Add both to appropriate array
            trustAnother[truster] = true;
            trustCount[trustee]++;
        }
        
        // Iterate trustCount and trustAnother array
        for (int i=1; i<=n; i++) {
            
            // If ith person doesn't trust anyone and everyone (n people) trust him, return i because that's the judge
            // NOTE: n - 1 is used because there are N people, including the judge. The town judge doesn't trust himself.
            if (!trustAnother[i] && trustCount[i] == n - 1) {
                return i;
            }
        }
        
        // Return -1 if no judge was found
        return -1;
    }
}

File: Find the Town Judge.py
# https://leetcode.com/problems/find-the-town-judge/

class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        
        # If list is empty, return -1
        if len(trust) == 0:
            return -1
        
        # If list only has one element, return 2nd element inside of only pair
        elif len(trust) == 1:
            return trust[0][1]
        
        # Create a list that holds a person and the people they trust
        ans = list()
        
        # Traverse list and add each person to ans
        for pair in trust:
            ans.append(pair[0])
        
        # Traverse ans list
        judge = 0
        ans.sort()
        for idx in range(len(ans)):
            
            # If current person isn't the same as the one before, then update judge
            if ans[idx] - 1 != ans[idx - 1] and idx == 0:
                judge = idx
                
            # If everyone trust each other, then there's no judge so return -1
            elif ans[idx] - 1 != ans[idx - 1] and idx:
                return -1
            
        return judge
